workflow/controller/cache/cache.go

```diff
@@ -0,0 +1,124 @@
+package cache
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"sync"
+
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/argoproj/argo/errors"
+
+	log "github.com/sirupsen/logrus"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+)
+
+var cacheKeyRegex = regexp.MustCompile("^[a-zA-Z0-9][-a-zA-Z0-9]*$")
```

Comment:
is this a common error - I'd be tempted to remove this check

Reply:
this is pretty common, I think it's worth it to validate on our end rather than defer the error checking to k8s and risk swallowing errors